Refactor add attachment functionality

Rosa-Fox opened the pull request:
Cleaning up the behaviour of attachments for Documents and Sections. We end up removing the `AttachmentUploader` class and simplifying various methods to make them easier to understand.
- 2nd PR as part of refactoring AttachmentUploader. The other is [here](https://github.com/alphagov/specialist-publisher-rebuild/pull/785).

Rosa-Fox committed:
Refactor add attachment functionality.

- Moves add_attachment method out of AttachmentUploader
- 2nd PR as part of refactoring AttachmentUploader. The other is [here](https://github.com/alphagov/specialist-publisher-rebuild/pull/785).

davidbasalla commented on app/models/document.rb:
```diff
@@ -251,6 +251,11 @@ def send_email_on_publish?
     update_type == "major"
   end
 
+  def add_attachment(document, attachment)
```
I think the document here refers to the class instance itself? So should the method be:

```
def add_attachment(attachment):
  attachments ||= []
  attachments.push(attachment)
end
```

benilovj replied:
fixed..

benilovj committed:
Remove unused method in Document

benilovj committed:
Simplify existing/new attachment tracking

benilovj committed:
Default attachments to be an empty array

davidbasalla committed:
Remove AttachmentUploader

After the previous commits, most functionality has been moved out of the AttachmentUploader class. This commit moves the remaining `upload` method to the Document and Section classes. This now groups all attachment related methods in one place which should help with future refactoring.

We also add unit tests around the 'upload' methods for Document, Section and Attachment classes.
